addons/hr_work_entry_contract/models/hr_work_entry.py

```diff
@@ -54,6 +55,34 @@ def _init_column(self, column_name):
                 WHERE _hwe.id = ANY(result.entry_ids)
             """)
 
+    def _get_missing_work_entry(self):
+        data = {}
+        work_entries_dict = {}
+        for work_entry in self:
+            key = (work_entry.date_start.date(), work_entry.employee_id)
+            if key not in work_entries_dict:
+                work_entries_dict[key] = []
+            work_entries_dict[key].append(work_entry)
+
+        for key, work_entries in work_entries_dict.items():
+            total_duration = sum(work_entry.duration for work_entry in work_entries)
+            if total_duration < work_entries[0].contract_id.resource_calendar_id.hours_per_day:
+                for work_entry in work_entries:
+                    data[work_entry] = total_duration
+        return data
+
+    @api.model
+    def _search_is_missing_work_entry_hours(self, operator, value):
+        if not isinstance(value, bool):
+            raise TypeError(_('Invalid value: %s') % value)
+        if operator not in ['=', '!=']:
+            raise TypeError(_('Invalid operator: %s') % operator)
+
+        work_entrys = self.search([])
+        new_operator = 'not in' if (operator == '=') ^ value else 'in'
+
+        return [('id', new_operator, [work_entry.id for work_entry in work_entrys._get_missing_work_entry().keys()])]
```

Comment:
If you do the change as suggested above, you could have this instead:
```suggestion
        return [('id', new_operator, work_entries._get_missing_work_entry().ids)]
```